### PR TITLE
Add a vLLM attention route for Qwen 3

### PR DESCRIPTION
**Pull request by anthony-etim**
## Description of the change

Adds a vLLM attention route to Qwen 3, so `Qwen3Attention` can serve through vLLM's TPU backend on the native flax/nnx path, following the routes already in #2794 for GPT-2, Llama, Qwen, and Gemma.

In `keras_hub/src/models/qwen3/qwen3_attention.py`, `call()` delegates to a new `_compute_vllm_attention` while a serving context is active and runs the original dense path otherwise. The route projects Q/K/V, applies RoPE at the engine's per-token positions, and calls the shared paged-attention bridge. The layer's `cache` argument comes back untouched: while serving, vLLM owns the paged KV cache, so the dense one is bypassed.

Qwen 3 normalizes queries and keys before RoPE. The route applies `_query_dense_layer_norm` and `_key_dense_layer_norm` in the dense path's order, and a test recomputes the expected tensors.

The kernel expands K/V for grouped-query attention itself, so the route passes the unexpanded `num_key_value_heads`.

## Tests

`keras_hub/src/vllm/qwen3_route_test.py` drives a real `Qwen3Attention` with a recording stand-in kernel, so it runs on CPU with no TPU or vLLM install:

- the route passes the unexpanded K/V head count, the right query head count, and the layer's own scale
- the query and key layer norms are applied before RoPE, checked against recomputed tensors
- the sliding window reaches the kernel when `sliding_window_size` is set, and is `None` otherwise
- off the serving path the layer's output is unchanged and the kernel is never called

Runs on the jax, tensorflow, and torch backends.

## Reference

Depends on keras-team/keras-hub#2794, which adds `keras_hub/src/vllm/` and the bridge this route calls; that should merge first. Companion backend PR: vllm-project/tpu-inference#3104.

## Checklist

- [x] I have added all the necessary unit tests for my change.
- [x] I have verified that my change does not break existing code and works with all backends.
- [x] My PR is based on the latest changes of the main branch.
- [x] I have followed the Keras Hub [Model contribution guidelines](https://github.com/keras-team/keras-hub/blob/master/CONTRIBUTING_MODELS.md).
- [x] I have followed the Keras Hub [API design guidelines](https://github.com/keras-team/keras-hub/blob/master/API_DESIGN_GUIDE.md).
- [x] I have signed the [Contributor License Agreement](https://cla.developers.google.com/about).

### CODE REVIEW

File: keras_hub/src/models/qwen3/qwen3_attention.py [hunk 1]
```diff
@@ -261,6 +274,36 @@ def _masked_softmax(self, attention_scores, attention_mask=None):
...
+
+        attention_output = vllm_paged_attention(
+            query,
+            key,
+            value,
+            self._inv_norm_factor,
+            num_kv_heads=self.num_key_value_heads,
+            sliding_window=self.sliding_window_size
+            if self.sliding_window_size
+            else None,
```
[the remark this answers is not in the document]
Comment (anthony-etim) [reply keras_hub/src/models/qwen3/qwen3_attention.py:300]: The dense path treats 0 as no window, so the check keeps the route doing the same.

**Comment by divyashreepathihalli** [comment]
@jules Can you verify the changes in this PR. Is this backward compatible and is the attention logic still preserved?

**Review by divyashreepathihalli — APPROVED** [review]
LGTM! thank you!
